packages/@eventual/core-runtime/src/local/stores/entity-store.ts

```diff
@@ -294,24 +308,19 @@ export class LocalEntityStore extends EntityStore {
      * Here we assume that the write operations are synchronous and that
      * the state of the condition checks will not be invalided.
      */
-    await Promise.all(
-      items.map(async (item) => {
-        if (item.operation === "put") {
-          return await this._put(
-            item.entity,
-            item.value,
-            item.key,
-            item.options
-          );
-        } else if (item.operation === "delete") {
-          return await this._delete(item.entity, item.key, item.options);
-        } else if (item.operation === "condition") {
-          // no op
-          return;
-        }
-        return assertNever(item);
-      })
-    );
+    items.forEach((item) => {
```

Comment:
For each?

Reply:
Aren't these promises?

Reply:
No, the problem was that they were promises and happening in "parallel" with other things (ex: other transactions).

So now the whole function is async (except for the response).